interactive/dashboard/api.py

```diff
@@ -0,0 +1,1009 @@
+import re
+from django.shortcuts import render
+from .forms.optionsforms import *
+from .forms.report_bug_form import BugForm
+from .forms.evolvingforms import *
+from .forms.initial_condforms import *
+from .flow_diagram import generate_flow_diagram
+from .flow_diagram import get_simulation_length, get_species
+from .upload_handler import *
+from .build_unit_converter import *
+from .save import *
+from .models import Document
+from django.http import HttpResponse, HttpResponsePermanentRedirect, Http404
+from django.http import JsonResponse, HttpResponseBadRequest, HttpRequest
+import os
+from django.conf import settings
+import mimetypes
+from django.core.files import File
+from interactive.tools import *
+import pandas
+import platform
+import codecs
+import time
+from io import TextIOWrapper
+from rest_framework import generics, status, views, permissions
+from rest_framework.response import Response
+from mechanism.reactions import *
+from mechanism.species import *
+from mechanism.network_plot import *
+from plots.plot_setup import *
+from dashboard.forms.formsetup import *
+from plots.plot_setup import *
+from os.path import exists
+from model_driver.session_model_runner import *
+from dashboard.flow_diagram import *
+import logging
+
+# api.py contains all DJANGO based backend requests made to the server
+# each browser session creates a "session_key" saved to cookie on client
+#       - request.session.session_key is a string representation of this value
+#       - request.session.session_key is used to access documents from DJANGO
+
+
+def beautifyReaction(reaction):
+    if '->' in reaction:
+        reaction = reaction.replace('->', ' → ')
+    if '__' in reaction:
+        reaction = reaction.replace('__', ' (')
+        reaction = reaction+")"
+    if '_' in reaction:
+        reaction = reaction.replace('_', ' + ')
+    return reaction
+
+
+# log, string2 is optional argument only for convenience
+def log(string, string2="", string3=""):
```

Comment:
@jwoody02 these passthrough functions are not acceptable. If the string is too long you can spill the string onto the next line. Python supports multiline strings.